include/meta/meta.hpp

```diff
@@ -2125,6 +2125,21 @@ namespace meta
             using min = defer<min, T, U>;
         }
 
+        /// Index of the first occurrence of the type \p T within the list \p List.
+        /// \par Complexity
+        /// \f$ O(N) \f$.
+        /// \ingroup list
+        template <typename T, typename List>
```

Comment:
ok, meta::nil_ is much better to return because of pattern matching and easier static_asserts … 


may I ask, why  the underscores at the end of  nil_ ?




Am 26.02.2015 um 00:01 schrieb gnzlbg <notifications@github.com>:

> In include/meta/meta.hpp:
> 
> > @@ -2125,6 +2125,21 @@ namespace meta
> >              using min = defer<min, T, U>;
> >          }
> >  
> > +        /// Index of the first occurrence of the type \p T within the list \p List.
> > +        /// \par Complexity
> > +        /// \f$ O(N) \f$.
> > +        /// \ingroup list
> > +        template <typename T, typename List>
> For which monad in particular did you wish and how would it have helped?
> 
> In meta there is only one data type, so I don't really know how useful a monad type class could be. I can see how Foldable is really useful, but again, for a single data type, it feels like over engineering.
> 
> IIRC Louis commented in one of his presentations that the Monad type class was "not super useful" in Boost.Hana, but you were there, so there might have been very interesting discussions after the presentation.
> 
> —
> Reply to this email directly or view it on GitHub.
>